container/set/set.go

```diff
@@ -45,6 +45,13 @@ type Set interface {
 
 // Returns a new set which is the union of s1 and s2.  s1 and s2 are unmodified.
 func Union(s1 Set, s2 Set) Set {
+	if s1 == nil || s2 == nil {
+		if s1 == nil {
+			return s2
```

Comment:
This is wrong.  We need to return s2.Copy() instead (same below).

Please fix internal repo as well ...